api/logic/environment_logic.go

```diff
@@ -85,7 +86,17 @@ func (e *L0EnvironmentLogic) CreateEnvironment(req models.CreateEnvironmentReque
 		return nil, errors.Newf(errors.MissingParameter, "EnvironmentName is required")
 	}
 
-	environment, err := e.Backend.CreateEnvironment(req.EnvironmentName, req.InstanceSize, req.MinClusterCount, req.UserDataTemplate)
+	if req.OperatingSystem == "" {
+		return nil, errors.Newf(errors.MissingParameter, "OperatingSystem is required")
+	}
+
+	environment, err := e.Backend.CreateEnvironment(
+		req.EnvironmentName,
+		req.InstanceSize,
+		req.OperatingSystem,
+		req.AMIID,
```

Comment:
[nit] I guess the guideline here (https://msdn.microsoft.com/en-us/library/141e06ef(v=vs.71).aspx) would be to use `AmiID`. although that looks even weirder